src/web/app/components/question-types/question-statistics/question-statistics-calculation/constsum-recipients-question-statistics-calculation.ts

```diff
@@ -55,6 +57,19 @@ export class ConstsumRecipientsQuestionStatisticsCalculation
       this.totalPointsPerOption[recipient] = sum;
       this.averagePointsPerOption[recipient] = +(answers.length === 0 ? 0 : sum / answers.length).toFixed(2);
     }
+
+    let totalSum: number = 0;
+    let totalStudents: number = 0;
+    for (const recipient of Object.keys(this.pointsPerOption)) {
+      totalSum += this.totalPointsPerOption[recipient];
+      totalStudents += 1;
+    }
+
+    for (const recipient of Object.keys(this.pointsPerOption)) {
+      this.averagePointsExcludingSelf[recipient] =
+          +(totalStudents === 0 ? 0 : (totalSum - this.totalPointsPerOption[recipient]) / totalStudents).toFixed(2);
+    }
```

Comment:
Not really sure what is being attempted here! Our goal is to find out who the 'self' is, and find a way to exclude their score from the total `sum`. This can be done by possibly comparing the response's recipient to the response's giver, an example of this can be found in say `rankPerOptionExcludeSelf` of `rank-recipients-question-statistics-calculation.ts`

Reply:
Apologies for not providing a better explanation in the original commit! What the above code snippet does is that it runs through each recipient, and takes the total of all student scores (`totalSum`), substracts the score of the recipient (`this.totalPointsPerOption[recipient]`) and divides it by all students (`totalStudents`) to obtain the average. 

while `rankPerOption` and `rankPerOptionExcludeSelf` can be computed simultaneously due to their similar nature, `Average` and `AverageExceptSelf` are slightly different as one is the average of one student's scores while the other is the average of all students' scores excluding one student...unless I am understanding something wrong here. Much apologies if I am!

it is because of this reason that I put the two computations in separate loops, the computation for `AverageExceptSelf` only takes place after the total of all students scores is computed.

Reply:
> `Average` and `AverageExceptSelf` are slightly different as one is the average of one student's scores while the other is the average of all students' scores excluding one student

From what I understand, 'average excluding self' might have a different meaning: it is the average of all scores received by _one_ recipient, excluding the recipient's own score (if applicable).

Suppose recipient Gene receives two scores: 100 from Gene (a self-review) and 50 from Charlie. Then the average points received should be 75, and the average points received excluding self should be 50, as it discounts the 100 from self-review.

But under your current scheme, the average excluding self returns 131.25:

<img width="1209" alt="Screenshot 2021-06-28 at 11 32 20 PM" src="https://user-images.githubusercontent.com/41856541/123663680-39557680-d869-11eb-8e21-135c2e43f8e5.png">

Reply:
Ah i understand what went wrong now! My bad, it seems like I misunderstood the meaning behind `Average Excluding Self`. That would explain the slight confusion: I have amended the code accordingly! Thanks so much for helping to clarify things!